### ISSUE TITLE
Consistency in list functions

### ISSUE THREAD

**Issue by m4dcoder** [harness/harness-go-sdk#107]
Looking at the following list functions, there are some inconsistencies in the input args. Some have filters while others not. Some filter use specific struct while another uses separate args. Most of the args start with limit and offset but then ListServicesByApplicationId starts with appId.

- ListDelegates(limit int, offset int)
- ListDelegatesWithFilters(limit int, offset int, name string, status graphql.DelegateStatus, delegateType graphql.DelegateType)
- ListCloudProviders(limit int, offset int)
- ListApplications(limit int, offset int)
- ListServices(limit int, offset int, filters []*graphql.ServiceFilter)
- ListServicesByApplicationId(appId string, limit int, offset int)

Like to propose the following changes.
- ListServicesByApplicationId(limit int, offset int, appId string)
- ListDelegatesWithFilters(limit int, offset int, filters[]*graphql.DelegateFilter)

**Comment by micahlmartin** [comment]
@m4dcoder yeah this all makes sense. Until I added the `ListServices` I hadn't figured out how to do the filters properly. Now that the pattern is established it should be easier to add this to the other functions.
